### PR TITLE
NAS-100209 / 12 / Add comment field for  initshutdown model

### PR DESCRIPTION
**Pull request by sonicaj**
This commit adds a migration and updates the initshutdown model to include a comment field.

### CODE REVIEW

File: gui/tasks/migrations/0014_ini_comment_field.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    dependencies = [
+        ('tasks', '0013_init_tasks_timeout'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='initshutdown',
+            name='ini_comment',
+            field=models.TextField(
```
Comment (william-gr) [inline gui/tasks/migrations/0014_ini_comment_field.py]: Not sure we want a textfield here, it seems to make sense to restrict the size of that field? Thoughts?

Comment (sonicaj) [reply gui/tasks/migrations/0014_ini_comment_field.py]: I did think of that initially, limiting to 100 chars perhaps as in tunables but then thought there might be some users who type longer comments ? But if you think that is better, updating the PR with that.

Comment (william-gr) [reply gui/tasks/migrations/0014_ini_comment_field.py]: Yes, a limit makes sense to me, it doesnt have to be as low as 100, could be 200

Comment (sonicaj) [reply gui/tasks/migrations/0014_ini_comment_field.py]: Perfect, i'll do that

File: src/middlewared/middlewared/plugins/init_shutdown_script.py [hunk 2]
```diff
@@ -22,6 +24,7 @@ class Config:
         Str('when', enum=['PREINIT', 'POSTINIT', 'SHUTDOWN'], required=True),
         Bool('enabled', default=True),
         Int('timeout', default=10),
+        Str('comment', default=''),
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/init_shutdown_script.py]: Range validator instead of that and COMMENT_LENGTH?

Comment (william-gr) [reply src/middlewared/middlewared/plugins/init_shutdown_script.py]: Ah, that doesnt work for Str, I guess?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/init_shutdown_script.py]: Good point, it doesn't but we should adapt it to work for string values as well checking their length if the value is str else integer it already is

**Comment by sonicaj** [comment]
I'll update it to 255 as well, let's save the extra byte ;)

**Pull request merged**